kernel/net/net_tcp.c

```diff
@@ -1808,6 +1809,48 @@ static int net_tcp_setsockopt(net_socket_t *hnd, int level, int option_name,
                 case SO_ERROR:
                 case SO_TYPE:
                     goto ret_inval;
+
+                case SO_RCVBUF:
+
+                    if(option_len != sizeof(uint32_t))
+                        goto ret_inval;
+
+                    tmp = sock->rcvbuf_sz;
+
+                    /* Maximum size for TCP window is currently 65535 */
+                    sock->rcvbuf_sz = (*(uint32_t *)option_value > 65535) ?
+                                      65535 : *(uint32_t *)option_value;
+
+                    new_ptr = realloc(sock->data.rcvbuf, sock->rcvbuf_sz);
+                    if(!new_ptr) {
+                        sock->rcvbuf_sz = tmp;
+                        goto ret_nomem;
+                    }
+
+                    sock->data.rcvbuf = new_ptr;
```

Comment:
It would make more sense to structure this such that the `tmp` variable is where the option value is copied into and that you only update the `sock->rcvbuf_sz` when you update `sock->data.rcvbuf` down below.

Also this needs a check for a sensible minimum value as well. Linux apparently enforces a minimum of 256 here (per the manpage), so let's do the same.